exporters/cr/exporter.go

```diff
@@ -0,0 +1,142 @@
+package cr
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8srestconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+var certManagerCertificateGroupVersionResource = schema.GroupVersionResource{
+	Group:    "cert-manager.io",
+	Resource: "certificates",
+	Version:  "v1",
+}
+
+type Config struct {
+	Namespaces []string
+}
+
+type Exporter struct {
+	certNotAfter  *prometheus.Desc
+	ctx           context.Context
+	logger        micrologger.Logger
+	dynamicClient dynamic.Interface
+
+	namespaces []string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Namespaces: []string{},
+	}
+}
+
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	e.logger.Log("info", "start collecting metrics")
+
+	namespacesToCheck := []string{}
+	// Create a list of namespaces to check
+	if len(e.namespaces) == 0 {
+		namespacesToCheck = append(namespacesToCheck, "")
+	} else {
+		namespacesToCheck = e.namespaces
+	}
+
+	listOptions := metav1.ListOptions{}
+
+	// Loop over namespaces
+	for _, namespace := range namespacesToCheck {
+		certs, err := e.dynamicClient.Resource(certManagerCertificateGroupVersionResource).Namespace(namespace).List(e.ctx, listOptions)
+		if err != nil {
+			e.logger.Log("error", microerror.Mask(err))
+			return
```

Comment:
I'll add a comment, I think `return` does make sense here, because if setting up/listing certificates will fail on one namespace, it will fail in the others too. 

Or do you think it will be possible that this will work other than I think?

Reply:
I was thinking that we maybe lack permissions in one namespace and then continue in the others. But you are likely correct and `return` makes more sense.